### PR TITLE
Issue #3667: Properly indent in format_xml_elements().

### LINKED ISSUE #3667: [DX] format_xml_elements() should use proper indentation
**Description of the bug**
The `format_xml_elements()` function takes an array and converts it into XML. This function tries but fails to use indentation on nested elements.

This came up as part of https://github.com/backdrop/backdrop-issues/issues/1992 (package signing) and in particular https://github.com/backdrop-contrib/project/issues/25, where update.module's XML feeds are being updated.

**Steps To Reproduce**
TBD

**Actual behavior**
Generated XML is indented a single space regardless of nesting. e.g.
```xml
<?xml version="1.0" encoding="utf-8"?>
<project xmlns:dc="http://purl.org/dc/elements/1.1/">
 <title>Webform</title>
 <short_name>webform</short_name>
 <releases>
 <release>
 <name>Webform 1.x-4.18.0</name>
 <version>1.x-4.18.0</version>
 <status>published</status>
 <date>1540427700</date>
 <filesize>286582</filesize>
 <security_update>false</security_update>
 <version_major>4</version_major>
 <version_minor>18</version_minor>
 <version_patch>0</version_patch>
 <release_link>https://github.com/backdrop-contrib/webform/releases/tag/1.x-4.18.0</release_link>
 <download_link>https://github.com/backdrop-contrib/webform/releases/download/1.x-4.18.0/webform.zip</download_link>
 </release>
 </releases>
</project>
```

**Expected behavior**
Generated XML should be indented two (or four) spaces per level of nesting. e.g.
```xml
<?xml version="1.0" encoding="utf-8"?>
<project xmlns:dc="http://purl.org/dc/elements/1.1/">
  <title>Webform</title>
  <short_name>webform</short_name>
  <releases>
    <release>
      <name>Webform 1.x-4.18.0</name>
      <version>1.x-4.18.0</version>
      <status>published</status>
      <date>1540427700</date>
      <filesize>286582</filesize>
      <security_update>false</security_update>
      <version_major>4</version_major>
      <version_minor>18</version_minor>
      <version_patch>0</version_patch>
      <release_link>https://github.com/backdrop-contrib/webform/releases/tag/1.x-4.18.0</release_link>
      <download_link>https://github.com/backdrop-contrib/webform/releases/download/1.x-4.18.0/webform.zip</download_link>
    </release>
  </releases>
</project>
```

---

PR https://github.com/backdrop/backdrop/pull/2576

### PR DESCRIPTION
**Pull request by quicksketch**
Fixes https://github.com/backdrop/backdrop-issues/issues/3667

### CODE REVIEW

**Comment by quicksketch** [comment]
Now updated with tests.

**Comment by jlfranklin** [comment]
Looks great to me.  Undoing those nested ternary conditionals makes the code a lot more readable, too.

File: core/modules/simpletest/tests/common.test [hunk 1]
```diff
@@ -2816,6 +2816,76 @@ class CommonFormatDateTestCase extends BackdropWebTestCase {
   }
 }
 
+/**
+ * Tests the format_xml_elements() API function.
+ */
+class CommonBackdropFormatXmlElementsUnitTestCase extends BackdropUnitTestCase {
+  /**
+   * Parse an example .info file an verify the results.
```
Comment (klonos) [inline core/modules/simpletest/tests/common.test]:
```suggestion
   * Parse an example .info file, and verify the results.
```

File: core/includes/common.inc [hunk 2]
```diff
@@ -2079,37 +2079,65 @@ function format_rss_item($title, $link, $description, $args = array()) {
...
- * entity-encoded or CDATA-escaped. Using this option is not recommended when
- * working with untrusted user input, since failing to escape the data
- * correctly has security implications.
+ *   If 'encoded' is TRUE it is up to the caller to ensure that 'value' is
+ *   either entity-encoded or CDATA-escaped. Using this option is not
+ *   recommended when working with untrusted user input, since failing to escape
+ *   the data correctly has security implications.
+ *
+ * @param int $indentation_level
+ *   Internal use-only indentation level. This function is called recursively
```
Comment (quicksketch) [inline core/includes/common.inc]:
```suggestion
 *   Internal-use only indentation level. This function is called recursively
```

File: core/modules/simpletest/tests/common.test [hunk 3]
```diff
@@ -2816,6 +2816,76 @@ class CommonFormatDateTestCase extends BackdropWebTestCase {
   }
 }
 
+/**
+ * Tests the format_xml_elements() API function.
+ */
+class CommonBackdropFormatXmlElementsUnitTestCase extends BackdropUnitTestCase {
+  /**
+   * Parse an example .info file, and verify the results.
```
Comment (quicksketch) [inline core/modules/simpletest/tests/common.test]:
```suggestion
   * Provide a sample data structure and verify the XML results.
```

**Comment by quicksketch** [comment]
Added `@since` docblock. After tests run again we're ready to merge here.

**Pull request merged**